Spiegel Online.js

```diff
@@ -39,151 +39,104 @@ http://www.spiegel.de/international/europe/0,1518,700530,00.html
-		title = title.split(" - ")[0];
-		newItem.title = title;
-	}
+	newItem.title = ZU.xpathText(doc, '//h2[@class="article-title"]') 
+		|| ZU.xpathText(doc, '//meta[@property="og:title"]/@content') 
+		|| ZU.xpathText(doc, '//title');
+	newItem.title = ZU.trimInternal(newItem.title);
+
 
 	// Tags
-	var tags_xPath = '//meta[contains(@name, "keywords")]';
-	var tags= doc.evaluate(tags_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext().content;
-	tags = tags.split(/,/);
-	tags = tags.slice(5); // The first six 5 Tags are generic or section info.
-	if (tags[0] != "" ) {
-		for (var i in tags) {
-			tags[i] = tags[i].replace(/^\s*|\s*$/g, '');
-			newItem.tags.push(tags[i]);
-		}
+	var tags = ZU.xpathText(doc, '//meta[contains(@name, "keywords")][1]/@content');
+	if (tags) {
+		// The first 5 Tags are generic or section info. TODO check if that is anymore true
+		newItem.tags = tags.trim().split(/(?:\s*,\s*)+/).slice(5);
 	}
 	
 	// Author
-	var author_XPath1 = ".//p[contains(@class, 'author')]"; // Most of the time, the author has its own tag. Easy Case, really.
-	var author_XPath2 =  ".//*[@id='spIntroTeaser']/strong/i"; // Sometimes, though, the author is in italics in the teaser.
-	if (doc.evaluate(author_XPath1, doc, null, XPathResult.ANY_TYPE, null).iterateNext()) {
-		var author = doc.evaluate(author_XPath1, doc, null, XPathResult.ANY_TYPE, null).iterateNext().textContent;
-		//Zotero.debug(author);	 
-	} else if  (doc.evaluate(author_XPath2, doc, null, XPathResult.ANY_TYPE, null).iterateNext()) {
-		var author = doc.evaluate(author_XPath2, doc, null, XPathResult.ANY_TYPE, null).iterateNext().textContent;
-		//Zotero.debug(author);	 
-	} else {
-		author = "";
-	}
-	author = author.replace(/^\s*By\s|^\s*(Ein.+?)?[Vv]on\s|\s*$/g, ''); // remove whitespace around the author and the "Von "at the beginning
-	if (doc.location.href.match(/^http\:\/\/www\.spiegel\.de\/spiegel/)){ // Spiegel Online and the Spiegel Archive have different formatting for the author line
-		author = author.split(/\sund\s|\su\.\s|\;\s|\sand\s/); 
-		for (var i in author) {
-			author[i] = author[i].replace(/(.*),\s(.*)/, '$2 $1');
+	var author = ZU.xpathText(doc, './/p[contains(@class, "author")]') // Most of the time, the author has its own tag. Easy Case, really.
+		|| ZU.xpathText(doc, '//*[@id="spIntroTeaser"]/strong/i'); // Sometimes, though, the author is in italics in the teaser.
+	if (author) {
+		author = author.replace(/^\s*By\s|^\s*(Ein.+?)?[Vv]on\s|\s*$/g, ''); // remove whitespace around the author and the "Von "at the beginning
+		if (url.indexOf('http://www.spiegel.de/spiegel/')>-1) { // Spiegel Online and the Spiegel Archive have different formatting for the author line
+			author = author.split(/\sund\s|\su\.\s|\;\s|\sand\s/); 
+			for (var i in author) {
+				author[i] = author[i].replace(/(.*),\s(.*)/, '$2 $1');
+			}
+		} else {
+			author = author.replace(/(,\s|in\s)\S*$|^\s*Aus.+?berichtet\s*/g, ""); //remove ", location" or "in location"
```

Comment:
you need to make sure that "in" doesn't match an end of somebody's name. Make it `\bin`